src/main/java/carbon/utils/Storage.java

```diff
@@ -0,0 +1,103 @@
+package carbon.utils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+import carbon.exceptions.InvalidFileFormatException;
+import carbon.task.Deadline;
+import carbon.task.Event;
+import carbon.task.Task;
+import carbon.task.TaskList;
+import carbon.task.Todo;
+
+/**
+ * Storage contains static methods that manage the storage of the TaskList.
+ */
+public class Storage {
+    private static final String TASKS_FILE_PATH = "data/user/tasks.txt";
+
+    /**
+     * Writes the tasks into storage.
+     *
+     * @param tasks The list of tasks.
+     */
+    public static void updateDataFile(ArrayList<Task> tasks) {
+        try (FileWriter writer = new FileWriter(TASKS_FILE_PATH)) {
+            for (Task task : tasks) {
+                writer.write(task.getStorageText() + "\n");
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Loads data from storage into the TaskList.
+     *
+     * @param taskList TaskList to load data into.
+     * @return Message describing any abnormalities (empty if none).
+     */
+    public static String loadDataFile(TaskList taskList) {
+        String message = "";
+
+        File dataFile = new File(TASKS_FILE_PATH);
+        if (!dataFile.getParentFile().exists() && !dataFile.getParentFile().mkdirs()) {
+            throw new RuntimeException("Unable to create data/user/ directory");
+        }
+        boolean fileExists;
```

Comment:
As per [convention given](https://se-education.org/guides/conventions/java/intermediate.html#variables), we could use verb+noun. 
```suggestion
        boolean hasFile;
```
Is "hasFile" clearer?

Reply:
nice catch, thanks! will correct it